src/v/raft/voter_priority_tracker.cc

```diff
@@ -12,20 +12,24 @@
 namespace raft {
 namespace {
 // zero priority doesn't allow node to become a leader
-inline constexpr voter_priority zero_voter_priority = voter_priority{0};
+
+[[maybe_unused]] inline constexpr voter_priority zero_voter_priority
+  = voter_priority{0};
 // 1 is smallest possible priority allowing node to become a leader
 inline constexpr voter_priority min_voter_priority = voter_priority{1};
+inline constexpr voter_priority not_ready_voter_priority = voter_priority{2};
 } // namespace
 
 voter_priority_tracker::voter_priority_tracker(
   raft::vnode self, bool is_ready_for_leader_election)
   : _self(self)
   , _replica_priority_override(
-      is_ready_for_leader_election ? std::nullopt
-                                   : std::make_optional(min_voter_priority)) {}
+      is_ready_for_leader_election
+        ? std::nullopt
+        : std::make_optional(not_ready_voter_priority)) {}
 
 void voter_priority_tracker::block_new_leadership() {
```

Comment:
we're not blocking new leadership here anymore, shall the method be renamed then?